pyqtgraph/graphicsItems/ViewBox/tests/test_ViewBox.py

```diff
@@ -76,10 +57,55 @@ def test_ViewBox():
     h = vb.geometry().height()
     view1 = QRectF(-5, 0, 20, 10)
     size1 = QRectF(0, h, w, -h)
-    assertMapping(vb, view1, size1)
+    _assert_mapping(vb, view1, size1)
+
+
+def _assert_mapping(vb, r1, r2):
+    assert vb.mapFromView(r1.topLeft()) == r2.topLeft()
+    assert vb.mapFromView(r1.bottomLeft()) == r2.bottomLeft()
+    assert vb.mapFromView(r1.topRight()) == r2.topRight()
+    assert vb.mapFromView(r1.bottomRight()) == r2.bottomRight()
+
+
+
+function_set = set([test_resize, test_wide_resize, test_tall_resize,
+                    test_aspect_ratio_constraint])
+                    
+@pytest.mark.parametrize('function', function_set)
+def setup_function(function):
+    print('\nsetting up function %s' % function)
+    global app, win, vb, QRectF
+    app = pg.mkQApp()
+    QRectF = pg.QtCore.QRectF
+    qtest = pg.Qt.QtTest.QTest
+    
+    win = pg.GraphicsWindow()
+    win.ci.layout.setContentsMargins(0,0,0,0)
+    win.resize(200, 200)
+    win.show()
+    vb = win.addViewBox()
     
+    # set range before viewbox is shown
+    vb.setRange(xRange=[0, 10], yRange=[0, 10], padding=0)
     
-if __name__ == '__main__':
-    import user,sys
-    test_ViewBox()
```

Comment:
I wouldn't worry about it. I usually rewrite these blocks to set up a particular state every time I need to fix something, so it's fine if it does nothing for now. I usually start with `py.test --pdb`, but sometimes this is not convenient, particularly because the pdb event loop does not play nicely with the PyQt interactive event loop.